crates/ruff/src/jupyter/notebook.rs

```diff
@@ -157,39 +167,14 @@ impl Notebook {
                             TextRange::default(),
                         ),
                         Category::Syntax | Category::Eof => {
-                            // Maybe someone saved the python sources (those with the `# %%` separator)
-                            // as jupyter notebook instead. Let's help them.
-                            let contents = std::fs::read_to_string(path).map_err(|err| {
```

Comment:
Wouldn't it be possible to keep this logic by rewinding the reader and then reading it to a string?

Reply:
No, because the reader gets consumed above by the Serde call I believe. I did try this!

Reply:
Maybe possible with `reader.by_ref()`, I will try again.